content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -0,0 +1,1384 @@
+- **diameterX** sets the diameter of the sphere's X axis and overrides any value in diameter. 
+- **diameterY** sets the diameter of the sphere's Y axis and overrides any value in diameter. 
+- **diameterZ** sets the diameter of the sphere's Z axis and overrides any value in diameter. 
+- **arc** is the revolution of the sphere on the Y axis on a 0-1 scale.
+- **slice** removes loops from the bottom of the sphere upward on a 0-1 scale.
+
+#### Outputs
+- **geometry** is the generated sphere and fulfills the requirement for at least one geometry in the graph.
+
+![Torus source node](/img/tools/nge/torusNode.jpg)
+
+### Torus
+This source node creates a torus of the given diameter at the resolution set by tessellation as follows.
+
+#### Inputs
+- **diameter** sets the diameter of the ring of the torus.
+- **thickness** sets the diameter of the circle defining the thickness of the torus.
+- **tessellation** sets the resolution for the torus both in diameter and thickness.
+
+#### Outputs
+- **geometry** is the generated torus and fulfills the requirement for at least one geometry in the graph.
+
+## Inputs
+Input nodes consist of basic value types used to set properties on other nodes. There are a few common properties on all input nodes. Those consist of the following.
+
+- **Mode** allows the input to be defined as a user-defined value or a contextual value of any type. User-defined values are just as they seem, the user assigned a constant value to the node. A contextual value is derived from the closest geometry node associated with this node. Changing mode to a different value can also change the type of data the node represents. For example, a Float input can be changed to a Vector2 contextual value which will change the data type supplied by the node. Be aware of this when changing mode as it can affect connections to other nodes. Note that if mode is set to a contextual value, the node will no longer display *Visible on frame*, *Min*, *Max*, or *Value* in the node properties but will instead display *Contextual value*.
+- **Contextual value** assigns the source of the data for any input set to a contextual value. The sources available depend on the data type selected for the input node. For example, choosing a Vector2 contextual value allows access to vertex IDs and choosing a Vector3 contextual value would offer vertex positions.
+- **Visible on frame** is a boolean which allows any input contained within a frame to be visible in the INPUTS list of the properties panel for a frame. Simply enable the boolean and then whenever a frame is selected - expanded or collapsed - any visible input can have its value changed directly in the frame properties panel.
+
+![Float input node](/img/tools/nge/FloatNode.jpg)
+
+### Float
+This input node represents a Float value rounded to four decimal places. The node contains the following properties which can be set in the properties panel. 

```

Comment:
I can't validate this one because entering more than 4 decimal places in the input field does round to 4 places for display in the property and on the node. But going back to edit the value again, I only see the rounded four decimal place value. Even if we do pass the more accurate number into the initial calculation, if the user comes back to update the value, won't it still get rounded? I think this one is more about what the user perceives since it does not appear that the input field does not appear to hold more than four decimal places. If it does, it seems like when we click in the input field to edit the value, it should display the full value for edit rather than the rounded value. Does that make sense?

Reply:
Yes, I think that for time being we can simply keep things as is and don't mention that it would be possible to type more digits.

Reply:
Sounds good. We can revisit this language if we make any changes to the UI to allow the user to see the exact value passed to the engine.